routes/chatRoutes.js

```diff
@@ -0,0 +1,81 @@
+const express = require("express");
+
+
+// NEW CONTROLLERS
+const Chat = require("../controllers/v1/chat");
+
+// MIDDLEWARE
+const authMiddleware = require("../middleware/auth");
+
+const router = express.Router();
+
+router.use(authMiddleware.protect);
+
+// SOCKET 
+
+// an array to track which users are online
+let onlineUsers = [];
+
+// add a users id and socket id to the onlineUsers array
+const addUser = (userId, socketId) => {
+  !onlineUsers.some((user) => user.userId === userId) && onlineUsers.push( {userId, socketId });
+};
+
+// remove a users id and socket id from the onlineUsers array
+const removeUser = (socketId) => {
+  onlineUsers = onlineUsers.filter((user) => user.socketId !== socketId);
+};
+
+// get a user's socket id from their user id
+const getUser = (userId) => {
+  return onlineUsers.find((user) => user.userId === userId);
+};
+
+// socket code
+const socketMiddleware = (req, res, next) => {
+  const io = req.app.get('socketio');
+
+  io.on("connection", (socket) => {
+    console.log("A user connected");
```

Comment:
add these whole socket lines (from lie 12 to line 67) in an external file, maybe in middlewares or helpers, and export all them by either using `modules.export` on each function or any other way you prefer!! then import the socketMiddleware in the chatRoutes file!